pkg/controller/service/reconciler.go

```diff
@@ -372,6 +398,8 @@ func (s *ServiceReconciler) Reconcile(ctx context.Context, id string) (result re
 		return
 	}
 
+	manifests = enforceNamespace(manifests, svc.Namespace, svc.SyncConfig)
```

Comment:
Isn't this actually built into cli utils?  I don't think we need to reinvent this logic.

The `cli-utils` implementation also throws an error which is probably the correct approach instead of silently filtering

Reply:
the `cli-utils` method `manifestreader.SetNamespaces` verifies that every namespaced resource has the namespace, and if one does not, it will set the namespace. The service namespace will throw the error if it already has a different one. I can't override the existing resource namespace with the service namespace

Reply:
Ok I see.  I also think it's likely we want to block creation of cluster-scoped resources entirely if this setting is enabled, which currently would be allowed

Reply:
OK, I have changed it, PTAL